### PR TITLE
stats: maintenance mode stat

### PR DESCRIPTION
**Pull request by junr03**
@lyft/network-team 

Expose stat for requests that are shed due to maintenance mode.

### CODE REVIEW

File: source/common/router/router.cc [hunk 1]
```diff
@@ -190,6 +190,7 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::HeaderMap& headers, bool e
     callbacks_->requestInfo().setResponseFlag(Http::AccessLog::ResponseFlag::UpstreamOverflow);
     chargeUpstreamCode(Http::Code::ServiceUnavailable, nullptr);
     Http::Utility::sendLocalReply(*callbacks_, Http::Code::ServiceUnavailable, "maintenance mode");
+    cluster_->stats().upstream_rq_maintenance_mode_.inc();
```
Comment (RomanDzhabarov) [inline source/common/router/router.cc]: there should be public doc update for this change

Comment (junr03) [inline source/common/router/router.cc]: yep, it is forthcoming. Just wanted to get out code first real quick

File: test/common/router/router_test.cc [hunk 2]
```diff
@@ -214,6 +214,10 @@ TEST_F(RouterTest, MaintenanceMode) {
   Http::TestHeaderMapImpl headers;
   HttpTestUtility::addDefaultHeaders(headers);
   router_.decodeHeaders(headers, true);
+  EXPECT_EQ(
+      1U,
```
Comment (RomanDzhabarov) [inline test/common/router/router_test.cc]: nit: but can you check for 0 in the case you do not expect stats to be set

**Comment by RomanDzhabarov** [comment]
lgtm, few minor comments

File: test/common/router/router_test.cc [hunk 3]
```diff
@@ -213,7 +213,15 @@ TEST_F(RouterTest, MaintenanceMode) {
 
   Http::TestHeaderMapImpl headers;
   HttpTestUtility::addDefaultHeaders(headers);
+  EXPECT_EQ(
```
Comment (mattklein123) [inline test/common/router/router_test.cc]: I think @RomanDzhabarov meant to put this on a different test case. I would put it on a case where a 503 is responded with that is not maintenance mode.

**Comment by RomanDzhabarov** [comment]
+1

**Pull request merged**